openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java

```diff
@@ -68,19 +67,25 @@ public OpenAPIConverterService() {
                 Optional<Path> documentPath = getPathFromURI(fileUri);
                 Optional<SyntaxTree> syntaxTree = workspaceManager.syntaxTree(documentPath.orElseThrow());
                 Optional<SemanticModel> semanticModel = workspaceManager.semanticModel(documentPath.orElseThrow());
-                Map<String, String> yamlContent =
-                        ServiceToOpenAPIConverterUtils.generateOAS3Definition(syntaxTree.orElseThrow(),
-                                semanticModel.orElseThrow(), null, false, null);
+
                 //Response should handle
+                if (semanticModel.isEmpty() || syntaxTree.isEmpty()) {
+                    StringBuilder errorString = getErrorMessage(syntaxTree, semanticModel);
+                    response.setYamlContent(errorString.toString());
```

Comment:
We need to return after setting the error message. Otherwise, the flow will continue and will throw NPE

Reply:
Could you please verify, is this the error message that is expected [https://github.com/ballerina-platform/ballerina-openapi/blob/ef108dba3f5154277929133453f7b8d3c5e51596/openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java#L84](https://github.com/ballerina-platform/ballerina-openapi/blob/ef108dba3f5154277929133453f7b8d3c5e51596/openapi-ls-extension/src/main/java/io/ballerina/openapi/extension/OpenAPIConverterService.java#L84), if we add the error field to the response there will it affect the old client? updated API changes are with this PR https://github.com/ballerina-platform/ballerina-openapi/pull/722 this will be available with beta4

Reply:
Does beta3 released version have the `setError` method? I am not sure how the extension should return the errors. If there is a standard way of doing it, we can do that.

Reply:
yeah,  In OpenAPI LS implementation, the error field was not available with beta3?

Reply:
Let's keep the Beta3 version. However, I think we need to return after setting the error right?

Reply:
Sorry for earlier I was confused with the error message representation. the suggested review was fixed by this commit [a47c1ee](url)